### ISSUE TITLE
Modified vertex property is not removed after remove operation

### ISSUE THREAD

**Issue by porunov** [JanusGraph/janusgraph#1981]
If the property is modified and then removed in the same transaction, it won't be removed in the storage backend.

For confirmed bugs, please report:
- Version: 0.5.0-SNAPSHOT
- Storage Backend: ScyllaDB 3.2.1 (not important)
- Mixed Index Backend: ElasticSearch 6.8.6 (not important)
- Mailing list Thread URL: https://groups.google.com/forum/#!topic/janusgraph-users/c4AAydzlEV4
- Steps to Reproduce: 

```
@Test
public void testRemoveVertexPropertyAfterModification() {

    Vertex v = graph.addVertex();
    v.property("name", "name1");
    graph.tx().commit();

    v = graph.traversal().V(v.id()).next();
    v.property("name", "name2");
    v.property("name").remove();
    graph.tx().commit();

    assertFalse(graph.traversal().V(v.id()).values("name").hasNext());
}
```

As seen from the example above, the `name` property isn't removed even so we called `remove` before the commit.

**Comment by porunov** [comment]
Current workaround is to call `remove` method 2 times for modified properties to be removed

**Comment by porunov** [comment]
Fixed in #2244
